lib/src/lints/avoid_returning_widgets/avoid_returning_widgets_rule.dart

```diff
@@ -92,11 +97,33 @@ class AvoidReturningWidgetsRule extends SolidLintRule {
 
       final isWidgetReturned = hasWidgetType(returnType);
 
+      final isIgnored = _hasIgnored(node, returnType);
+
       final isOverriden = node.declaredElement?.hasOverride ?? false;
 
-      if (isWidgetReturned && !isOverriden) {
+      if (isWidgetReturned && !isOverriden && !isIgnored) {
         reporter.reportErrorForNode(code, node);
       }
     });
   }
+
+  bool _hasIgnored(Declaration node, DartType returnType) {
+    final methodName = node.declaredElement?.name;
+
+    final excludedItem = config.parameters.exclude
+        .firstWhereOrNull((e) => e.methodName == methodName);
+
+    if (excludedItem
+        case AvoidReturningWidgetsExclude(
+          :final String? className,
+        )) {
+      if (className == null) {
+        return true;
+      } else {
+        return returnType.hasIgnoredType(ignoredTypes: {className});
+      }
+    }
+
+    return false;
```

Comment:
```suggestion
    final excludedItem = config.parameters.exclude
        .firstWhereOrNull((e) => e.methodName == methodName);
   
   if (excludedItem == null) return false;
   
   final className = excludedItem.className;
   
   if (className == null) return false;
   
   return returnType.hasIgnoredType(ignoredTypes: {className});
```

I would refactor this part into something like this to make it a bit more readable. What do you think?